### PR TITLE
New alpha calculator: Dither

### PR DESCRIPTION
**Pull request by serin-delaunay**
Implements one of the items in #37. With the information currently available to the AlphaCalculator (weights, no pixel location, local evaluation), only uniform random dithering is possible. It derives from `CalculatorTopTwo` to re-use code.

It looks good on visual tests. Uses a lower power than CalculatorTopTwo in order to make the dithered area larger.

TODO:
- [x] Add to CMakeLists.txt.
- [x] Add tests.
- [x] Update options schema.
- [x] When merging, remove note in `doc/algorthm.md` about Dither being unavailable.
- [x] Conflict resolution broke compilation, fix it.

### CODE REVIEW

**Comment by hryniuk** [comment]
You've got a TODO here :wink: (and a conflict, but I'm going to check other PRs, so the `master` branch will be updated soon).

**Comment by serin-delaunay** [comment]
Yep :) I'm not in a rush to merge this branch, I think it can wait until after https://github.com/Wangscape/Wangscape/milestone/1.

File: WangscapeTest/TestCalculatorDither.cpp [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+            {
+                if (i == winner_index)
+                    EXPECT_EQ(255, alphas[i]);
+                else
+                    EXPECT_EQ(0, alphas[i]);
+            }
+        }
+        if (expected_winners)
+        {
+            if (cd.power < 10)
```
Comment (hryniuk) [inline WangscapeTest/TestCalculatorDither.cpp]: Why `10` is special? Could you name it?

File: WangscapeTest/TestCalculatorDither.cpp [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        validateResult(cd1, weights, std::make_pair(weight, weight_runner_up));
+        validateResult(cd2, weights, std::make_pair(weight, weight_runner_up));
+        validateResult(cd256, weights, std::make_pair(weight, weight));
+    }
+}
+
+TEST_F(TestCalculatorDither, TestCalculatorDitherRandom)
+{
+    std::uniform_real_distribution<double> weight_gen(-10, 10);
+    for (int i = 0; i<20; i++)
```
Comment (hryniuk) [inline WangscapeTest/TestCalculatorDither.cpp]: Why `20`?

Comment (serin-delaunay) [inline WangscapeTest/TestCalculatorDither.cpp]: It's just an arbitrary number of times to run the randomised test. `TestCalculatorDither::validateResult` has its own loop, so I didn't want to make this number too high. Currently this test calls `updateAlphas` 5120 times.

File: WangscapeTest/TestCalculatorDither.cpp (hunk 1 above)
Comment (serin-delaunay) [inline WangscapeTest/TestCalculatorDither.cpp]: This line separates `cd1` and `cd2` from `cd256`, which has very different behaviour. `10` could instead be anything from `2+epsilon` to `256`.

Comment (hryniuk) [inline WangscapeTest/TestCalculatorDither.cpp]: Place here at least a comment about it :).

Comment (serin-delaunay) [inline WangscapeTest/TestCalculatorDither.cpp]: Done.

**Pull request merged**